index.js

```diff
@@ -13,7 +15,8 @@ const replacer = (match, group) => {
 };
 
 class WhiteOut extends Stream.Transform {
-  constructor (rules, options) {
+  constructor (rules, root, options) {
+    Assert.ok(root == null || typeof root === 'string', 'root must be a string'); // eslint-disable-line eqeqeq
```

Comment:
I was thinking you'd pass `null` if you want to set `options` and not `root`. If you want to set something to "not set" I usually go with `null`, not setting to `undefined`.